docs/developer_documentation/ref-permissions-api.md

```diff
@@ -0,0 +1,714 @@
+- Users inherit permissions from their roles, and roles from their parentroles.
+- The access control list (ACL) for a resource can have a parent, in that case permissions on the parent also grant permissions on the child ACL's. In the current MOLGENIS system this is the case for entity types and packages, both inherit permissions from the package in which they reside..
+
+## Managing row level security
+### Getting all resource types in the system
+##### Endpoint
+```
+GET https://molgenis.mydomain.example/api/permissions/types
+```
+##### Parameters
+URL: none
+Query: none
+
+##### Response
+| Status code         | Description                                                                       |
+|---------------------|-----------------------------------------------------------------------------------|
+|200 Ok | Success |
+|403 Forbidden | the user has no permissions to execute this action |
+
+##### Response body
+List of ACL types available in the system.
+##### Example 
+Response:
+```
+[
+    {
+        "id": "entity-sys_ImportRun",
+        "entityType": "sys_ImportRun",
+        "label": "Import"
+    },
+    {
+        "id": "entity-sys_job_ResourceCopyJobExecution",
+        "entityType": "sys_job_ResourceCopyJobExecution",
+        "label": "Resource Copy Job Execution"
+    },
+    {
+        "id": "entity-hospital_cardiology_patients",
+        "entityType": "hospital_cardiology_patients",
+        "label": "patients"
+    },
+    {
+        "id": "package",
+        "entityType": "sys_md_Package",
+        "label": "Package"
+    },
+    {
+        "id": "entity-sys_FileMeta",
+        "entityType": "sys_FileMeta",
+        "label": "File metadata"
+    },
+    {
+        "id": "plugin",
+        "entityType": "sys_Plugin",
+        "label": "Plugin"
+    },
+    {
+        "id": "entityType",
+        "entityType": "sys_md_EntityType",
+        "label": "Entity type"
+    }
+]
+   ```
+
+### Getting all suitable permissions for a resource type
+##### Endpoint
+```
+GET https://molgenis.mydomain.example/api/permissions/types/permissions/{typeId}
+```
+##### Parameters
+URL: TypeId as described in the [parameters section](##Parameters)
+
+##### Response
+| Status code         | Description                                                                       |
+|---------------------|-----------------------------------------------------------------------------------|
+|200 Ok | Success |
+|403 Forbidden | the user has no permissions to execute this action |
+|404 Not found | the type does not exist |
+
+##### Response body
+A list of permissions that can be used for this resource type.
+
+##### Example 
+Request:
+```https://molgenis.mydomain.example/api/permissions/types/permissions/entityType```
+Response:
+```[
+       "READMETA",
+       "READ",
+       "COUNT",
+       "WRITEMETA",
+       "WRITE"
+   ]
+   ```
+
+### Creating a new resource type in the system (Row level securing an entity)
```

Comment:
add: feature enabled only for superusers

Reply:
Disabling: ```This will delete a type, the access control lists for all rows will remain in the system.```. Could it be that enabling uses this information and only sets the current user as owner for rows that were added after disabling?